app/hooks/useTrucoGame.ts

```diff
@@ -218,6 +232,79 @@ export function useTrucoGame() {
     }
   }, [gameState]);
 
+  const handleEnvido = useCallback(async () => {
+    // Can only offer envido in the first turn of a round and before truco
+    if (gameState.phase.type !== 'HUMAN_TURN' || 
+        gameState.playedCards.length > 0 || 
+        gameState.trucoState.type !== 'NONE' ||
+        gameState.envidoState.type !== 'NONE') return;
+    
+    // Cannot offer envido when someone has flor
+    const humanHasFlor = hasFlor(gameState.humanCards, gameState.muestraCard);
+    const aiHasFlor = hasFlor(gameState.aiCards, gameState.muestraCard);
+    
+    if (humanHasFlor || aiHasFlor) {
+      setGameState(prev => ({
```

Comment:
If we make sure the Envido button is not shown in this situation, it shouldn't be a reachable path. We can log an error instead